Source/Core/arrayFill.js

```diff
@@ -0,0 +1,64 @@
+/*global define*/
+define([
+        './DeveloperError',
+        './defaultValue',
+        './defined'
+    ], function(
+        DeveloperError,
+        defaultValue,
+        defined) {
+    'use strict';
+
+    /**
+     * Fill an array or a portion of an array with a given value.
+     * @private
+     *
+     * @param {Array} array The array to fill.
+     * @param {*} value The value to fill the array with.
+     * @param {Number} start The index to start filling at.
+     * @param {Number} end The index to end stop at.
+     *
+     * @returns {Array} The resulting array.
+     */
+    function arrayFill(array, value, start, end) {
+        //>>includeStart('debug', pragmas.debug);
+        if (!defined(array) || typeof array !== 'object') {
+            throw new DeveloperError('Array is required.');
+        }
+        if (!defined(value)) {
+            throw new DeveloperError('Value is required.');
+        }
+        if (defined(start) && (typeof start !== 'number')) {
```

Comment:
No need for the defined here and below, because `typeof` will never be `number` if the value is undefined.